pkg/excmd/executor.go

```diff
@@ -34,19 +34,37 @@ func (e *executor) Exec(ctx context.Context, name string, opts ...Option) (out [
 	cmd.Dir = c.Dir
 	cmd.Env = c.Env
 
-	sigCh := make(chan os.Signal, 1)
-	signal.Notify(sigCh)
+	cCtx, cancel := context.WithCancel(context.Background())
+
 	wg.Add(1)
 
 	go func() {
 		defer wg.Done()
-		defer recover()
-		for sig := range sigCh {
-			zap.L().Debug("signal received", zap.Stringer("signal", sig))
-			if cmd.ProcessState == nil || cmd.ProcessState.Exited() {
-				break
+		select {
+		case <-ctx.Done():
+			cmd.Process.Signal(os.Interrupt)
+		case <-cCtx.Done():
+			// no-op
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		sigCh := make(chan os.Signal, 1)
+		signal.Notify(sigCh)
+
+		for {
+			select {
+			case sig := <-sigCh:
+				zap.L().Debug("signal received", zap.Stringer("signal", sig))
+				cmd.Process.Signal(sig)
```

Comment:
**[golangci]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Error return value of `cmd.Process.Signal` is not checked (errcheck)